src/marqo/core/unstructured_vespa_index/unstructured_add_document_handler.py

```diff
@@ -100,27 +102,29 @@ def _validate_field(self, field_name: str, field_content: Any) -> None:
                     f"Allowed content types: {[ty.__name__ for ty in ALLOWED_UNSTRUCTURED_FIELD_TYPES]}"
                 )
 
-            if isinstance(field_content, list) and not list_types_valid(field_content):
-                raise AddDocumentsError(
-                    f"Field content '{field_content}' "
-                    f"of type {type(field_content).__name__} is not of valid content type! "
-                    f"All list elements must be of the same type and that type must be int, float or string"
-                )
+            if isinstance(field_content, list):
+                for element in field_content:
+                    if not isinstance(element, str):
+                        # if the field content is a list, it should only contain strings.
+                        raise AddDocumentsError(
+                            f"Field content {field_content} includes an element of type {type(element).__name__} "
+                            f"which is not a string. Unstructured Marqo index only supports string lists."
+                        )
+
+            is_tensor_field = field_name in self.add_docs_params.tensor_fields
 
             if isinstance(field_content, dict):
                 if self.tensor_fields_container.is_custom_tensor_field(field_name):
-                    # TODO is_non_tensor_field check can be move out to AddDocsParams level
-                    validate_custom_vector(field_content, False, self.marqo_index.model.get_dimension())
-                elif self.tensor_fields_container.is_multimodal_field(field_name):
-                    # FIXME, multimodal field should not be present in the doc
-                    # TODO This validation should be done at AddDocsParams level
-                    validate_multimodal_combination(field_content, False,
-                                                    self.tensor_fields_container.get_multimodal_field_mapping(field_name))
-                elif self.marqo_index.parsed_marqo_version() < semver.VersionInfo.parse("2.9.0"):
-                    # TODO This check should not happen at root level
+                    # TODO should is_non_tensor_field check be moved out to AddDocsParams validation?
+                    # Please note that if one of the documents in the batch has a custom field which does not exist
+                    # in the tensor field, the whole batch will fail and user will get a 400 pydantic.ValidationError.
+                    # We keep this behaviour unchanged to be compatible with the legacy unstructured index.
+                    validate_custom_vector(field_content, not is_tensor_field, self.marqo_index.model.get_dimension())
```

Comment:
nit should we start moving things out of validation to decommission the module? Seems like bad coupling of the two index types

Reply:
Yes, this will be done in a follow-up refactoring.